keras_core/layers/preprocessing/discretization_test.py

```diff
@@ -64,9 +64,9 @@ def test_correctness(self):
         layer = layers.Discretization(
             bin_boundaries=[0.0, 0.5, 1.0], output_mode="count"
         )
-        output = layer(np.array([[0.1, 0.8, 0.9]]))
+        output = layer(np.array([0.1, 0.8, 0.9]))
         self.assertTrue(backend.is_tensor(output))
-        self.assertAllClose(output, np.array([[0, 1, 2, 0]]))
```

Comment:
The layer is expected to work for 2D inputs like these. We need to fix this case too.

Reply:
done!